docs/design.md

```diff
@@ -0,0 +1,279 @@
+# Indexer Reference Provider Design
+The goal of this reference implementation is to flesh out all the protocol details
+for data providers to be able to interact with indexer nodes. 
+
+## Overview
+- Data providers keep a local index of all the data they are providing.
+- Data providers are responsible for the advertisement of updates related to the data they are providing.
+        - Data providers must be able to uniquely identify the batches of data they are providing, and/or the
+        announcements they broadcast to the network in order to be able to serve indexer-node data ingestion requests.
+- Indexer nodes listen to advertisement from data providers. When they come across a previously unseeend
+advertisement for a data provider, they trigger a new ingestion session to sync with the provider. 
+
+## Data Structures
+Data providers and indexer nodes can use any internal representation they want for indexed data. They only
+data structures enforced by the protocol are in the interaction between the two.
+
+### Advertisements
+Advertisements are used by data providers to share updates about the data they are providing. Ideally,
+these advertisements are published in a shared pub-sub channel between providers and indexer nodes. However,
+some indexer nodes may choose to provide a dedicated endpoint to allow certain authenticated providers to push announcements and updates directly to them.
+- `Pull` Advertisements: Indexer-nodes pull advertisement from a shared pub-sub channel.
```

Comment:
`Pull` IMO is not the correct description, because the indexer is not initiating the action.  I would call this `Subscribe` if looking at it from the indexer side, or `Publish` if looking at it from the provider side.

There is an additional mechanism where the indexer does initiate action and asks the provider directly for the latest advertisement.  I would call that `Poll` or `Pull`.  This is discussed [here](https://www.notion.so/protocollabs/Indexer-Node-Design-4fb94471b6be4352b6849dc9b9527825#edfdbffea3d1407ba7ac696d490f0b3a) in the design doc.

Reply:
You're totally right. Adding it.